Functions/Import/xASL_imp_StudyParPriority.m

```diff
@@ -0,0 +1,90 @@
+%                 First context has the lowest priority, following ones are more important and review previous variables. 
+%                 The alias hierarchy within each context contains a list of regexp to be matched against subject/session/run.
+%                 If subject doesn't exist or regexp doesn't exist or aliasHierarchy doesn't exist, it allows it.
+%
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% EXAMPLE:        studyPar = xASL_imp_StudyParPriority(studyParFull, 'sub-01', '1', '2')
+% __________________________________
+% Copyright 2015-2022 ExploreASL
+
+% Check existence of studyParFull and return if single context
+if nargin < 1 || isempty(studyParFull)
+	studyParSpecific = struct();
+	return;
+end
+
+% Do nothing for a single-context studyPar
+if ~isfield(studyParFull,'StudyPars')
+	studyParSpecific = studyParFull;
+	return;
+end
+
+% Initialize empty studyPar and then go through different contexts
+studyParSpecific = struct();
+
+% At each step, check for all conditions and set a flag for overwrite (first step having the lowest priority
+for iContext = 1:length(studyParFull.StudyPars)
+	bOverwrite = 1;
+	
+	% Checks for fields SubjectRegexp, VisitRegexp, SessionRegexp - if missing or empty, then take this context into account
+	% Otherwise check if the respective field and Subject/Visit/Session name matches the regular expression
+	
+	% Check subject name - missing rule or missing regexp means that it is OK
+	if isfield(studyParFull.StudyPars{iContext},'SubjectRegexp')
+		if ~isempty(subjectName) && ~isempty(studyParFull.StudyPars{iContext}.SubjectRegexp)
+			% If the regexp is not found, then we don't take this subejct
+			if isempty(regexpi(subjectName, xASL_num2str(studyParFull.StudyPars{iContext}.SubjectRegexp)))
+				bOverwrite = 0;
+			end
+		end
+		studyParFull.StudyPars{iContext} = rmfield(studyParFull.StudyPars{iContext},'SubjectRegexp');	
+	end
+	
+	% Check sessions
+	if isfield(studyParFull.StudyPars{iContext},'VisitRegexp')
+		if ~isempty(sessionName) && ~isempty(studyParFull.StudyPars{iContext}.VisitRegexp)
+			% If the regexp is not found, then we don't take this subejct
+			if isempty(regexpi(sessionName, xASL_num2str(studyParFull.StudyPars{iContext}.VisitRegexp)))
+				bOverwrite = 0;
+			end
+		end
+		studyParFull.StudyPars{iContext} = rmfield(studyParFull.StudyPars{iContext},'VisitRegexp');	
+	end
+		
+	% Check runs
+	if isfield(studyParFull.StudyPars{iContext},'SessionRegexp')
+		if ~isempty(runName) && ~isempty(studyParFull.StudyPars{iContext}.SessionRegexp)
+			% If the regexp is not found, then we don't take this subejct
+			if isempty(regexpi(runName, xASL_num2str(studyParFull.StudyPars{iContext}.SessionRegexp)))
+				bOverwrite = 0;
+			end
+		end
+		studyParFull.StudyPars{iContext} = rmfield(studyParFull.StudyPars{iContext},'SessionRegexp');	
```

Comment:
Because - all is then passed to the JSON-BIDSify and we risk to write these to the JSON by mistake. It shouldn't happen, but I'd like to keep the code clean and remove things that have nothing to do there.

Reply:
OK, but then this should happen only at the end of the code, before it is passed to JSON-BIDSify. Now it limits debugging, because you do this for each subject/session/run level specifically.

Reply:
I don't see the issue. this function is supposed to take studyParFull and return a clean studyPar with a single instance. It has to be removed inside this function.

Reply:
Isn't outputting to a different variable better programming? Because that makes for easier bugfixing. So instead of returning the clean studyPar to the same parameter, you put it in `studyParClean`, and at the end of the iterations you overwrite `x.studyPar` by `studyParClean`.

Reply:
Sure. Solved by not touching studyPar anymore and cleaning the keywords at once in the final studyParSpecificSubjVisitSess
8ebcd98b35fa2b848d646952c1ce2cd3852d7f86